src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -190,15 +191,32 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     catch (Exception ex)
```

Comment:
>Exception [](http://example.com/codeflow?start=27&length=9)

The change here is limiting when the next refresh can be requested rather than performing another request. 

Have we discussed the possibility of doing the retry in here?

Reply:
IIRC the initial fetch already has a retry so that would be multiple levels of retries. That's not to say we shouldn't do this but my preference would be to have that as a separate change.